internal/exporters/base.py

```
@@ -0,0 +1,254 @@
+        """
+        operation = RegexCopyOperation(pattern, target_folder, keep_original_name, 
+                                     rename_func, custom_func, additional_sources)
+        self.operations.append(operation)
+    
+    def add_external_file_operation(self, external_path: str, target_path: str) -> None:
+        """Add an external file copy operation"""
+        operation = ExternalFileOperation(external_path, target_path)
+        self.operations.append(operation)
+    
+    def export(self, create_zip: bool = True) -> None:
+        """Export folder format"""
+        
+        name_length = max(len(operation.target_name()) for operation in self.operations) + 2
+
+        self.formatter.println(f"Exporting {self.output_path}...")
+        
+        # Create temporary directory for conversion
+        with tempfile.TemporaryDirectory() as temp_dir:
+            if not create_zip:
+                output_dir = Path(self.output_path)
+                if output_dir.exists():
+                    self.formatter.println(formatter.ANSI_RED, f"Error: path {output_dir} already exists.", formatter.ANSI_RESET)
+                    return
+                output_dir.mkdir();
+            else:
+                output_dir = Path(temp_dir)
+
+            try:
+                # Execute all operations
+                for operation in self.operations:
+                    self.formatter.print(" " * 4)
+                    self.formatter.print_fixed_width(operation.target_name(), width=name_length)
+                    operation.execute(self.formatter, self.context, output_dir)
+                
+                # Handle output
+                if create_zip:
+                    self.formatter.println(f"Creating zip file...")
+                    zip_path = Path(self.output_path)
+                    
+                    with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
```

Maybe we can throw an error if `zip_path` is already a file (e.g. when user calls `tmt export problem.yaml`).
FYR, there is [shutil.make_archive](https://docs.python.org/3/library/shutil.html#shutil.make_archive)

---

Your right.